packages/library-base/src/androidMain/kotlin/io/realm/kotlin/internal/platform/PRNGFixes.kt

```diff
@@ -0,0 +1,358 @@
+/*
+ * Copyright 2022 Realm Inc.
+ * Copyright 2017-2019 Carlos Ballesteros Velasco and contributors
+    * https://github.com/korlibs/korge/graphs/contributors
+    * https://github.com/korlibs-archive/
+ * All rights reserved.
+
+ * MIT License
+
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is furnished to do
+ * so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+
+ * See: https://github.com/korlibs/korge/tree/main/krypto
+*/
+
+package io.realm.kotlin.internal.platform
+
+import android.graphics.Bitmap
+import android.os.Build
+import android.os.Process
+import android.util.Log
+import java.io.ByteArrayOutputStream
+import java.io.DataInputStream
+import java.io.DataOutputStream
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.IOException
+import java.io.OutputStream
+import java.io.UnsupportedEncodingException
+import java.lang.StringBuilder
+import java.security.NoSuchAlgorithmException
+import java.security.Provider
+import java.security.SecureRandom
+import java.security.SecureRandomSpi
+import java.security.Security
+
+/**
+ * Fixes for the output of the default PRNG having low entropy.
+ *
+ * The fixes need to be applied via [.apply] before any use of Java
+ * Cryptography Architecture primitives. A good place to invoke them is in the
+ * application's `onCreate`.
+ */
+@Suppress("TooGenericExceptionThrown", "MagicNumber")
+object PRNGFixes {
```

Comment:
Seems like the motivation behind PRNGFixes (and the actual code snippet) is taken directly from [Android Dev Blogs](https://android-developers.googleblog.com/2013/08/some-securerandom-thoughts.html). I am not exactly well-versed in cryptographic things and I am unsure whether the UUIDs would need to be cryptographically random for security purposes (I'd assume not considering they have not been), so perhaps including something like that is an overkill and maybe even a bad idea? I also [found an issue where it might lead to some errors with some devices](https://stackoverflow.com/questions/19767349/android-securerandom-fix-prngfixes-and-prevention-infomation-message) so I removed it for now to see how it is like with CI.